extensions/json-language-features/client/src/jsonClient.ts

```diff
@@ -471,6 +470,22 @@ export async function startClient(context: ExtensionContext, newLanguageClient:
 		}
 	}
 
+	async function getSortTextEdits(document: TextDocument, tabSize: string | number = 4, insertSpaces: string | boolean = true): Promise<TextEdit[]> {
+		const filesConfig = workspace.getConfiguration('files', document);
+		const options: SortOptions = {
+			tabSize: Number(tabSize),
+			insertSpaces: Boolean(insertSpaces),
+			trimTrailingWhitespace: filesConfig.get<boolean>('trimTrailingWhitespace'),
+			trimFinalNewlines: filesConfig.get<boolean>('trimFinalNewlines'),
+			insertFinalNewline: filesConfig.get<boolean>('insertFinalNewline'),
+		};
+		const params: DocumentSortingParams = {
+			uri: document.uri.toString(),
+			options
+		};
+		return client.sendRequest(DocumentSortingRequest.type, params);
```

Comment:
The vscode API expects that we give it instances of `vscode.TextEdit`, not just other objects which, using duck-typing, match the same shape as `vscode.TextEdit`. Let's map these JSON objects here to instances of `vscode.TextEdit`.